### PR TITLE
Id check

### PR DESCRIPTION
**Pull request by jmhauck**
#1048

### CODE REVIEW

File: packages/common/src/featureServiceHelpers.ts [hunk 1]
```diff
@@ -1100,9 +1100,25 @@ export function _isSelfReferential(
...
+          if (srcTables[name].indexOf(id) > -1) {
+            return true;
+          } else {
+            srcTables[name].push(id);
+          }
+        } else {
+          srcTables[name] = [id]
+        }
+      }
+      return (table.relatedTables || []).some(r => names.indexOf(r.name) > -1);
```
Comment (MikeTschudi) [inline packages/common/src/featureServiceHelpers.ts:1120]: Same question here. `some` returns truthy or falsy.  If `some` were to return the falsy value "0", this condition would still return true.

File: packages/common/src/featureServiceHelpers.ts [hunk 2]
```diff
@@ -1100,9 +1100,25 @@ export function _isSelfReferential(
...
   const names = layersAndTables.map(l => l.item.name);
+  const srcTables = {};
   return layersAndTables.some(l => {
-    const relatedTables = l.item.adminLayerInfo?.viewLayerDefinition?.table?.relatedTables || [];
-    return relatedTables.some(r => names.indexOf(r.name) > -1);
+    const table = l.item.adminLayerInfo?.viewLayerDefinition?.table;
+    if (table) {
+      const name = table.sourceServiceName;
+      const id = table.sourceLayerId;
+      if (name && id > -1) {
```
Comment (MikeTschudi) [inline packages/common/src/featureServiceHelpers.ts:1109]: Why not `if(name && id) {`?

**Review by MikeTschudi — CHANGES_REQUESTED** [review]

File: packages/common/src/featureServiceHelpers.ts (hunk 1 above)
Comment (jmhauck) [reply packages/common/src/featureServiceHelpers.ts:1120]: indexOf(r.name) > -1 will return boolean

File: packages/common/src/featureServiceHelpers.ts (hunk 2 above)
Comment (MikeTschudi) [reply packages/common/src/featureServiceHelpers.ts:1109]: Thank you--I got operator precedence wrong

File: packages/common/src/featureServiceHelpers.ts (hunk 1 above)
Comment (MikeTschudi) [reply packages/common/src/featureServiceHelpers.ts:1120]: Thank you--I'd misread the line

**Review by MikeTschudi — APPROVED** [review]

**Pull request merged**